Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/insightidr.py

```diff
@@ -0,0 +1,836 @@
+    """
+    Returning 'ok' indicates that the integration works like it is supposed to.
+
+    200 - success
+    401 - API key not valid
+    500 - not account region
+    Args:
+        client(Client): Rapid7 Client
+
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+    try:
+        response = client.validate()
+        status_code = response.status_code
+        if status_code == 200:
+            return 'ok'
+
+        if status_code == 401:
+            return 'API key is not valid.'
+
+        if status_code == 500:
+            return 'This isn\'t your account region.'
+
+        return 'Something went wrong...'
+    except DemistoException:
+        return 'Connection error. Check your region.'
+
+
+def fetch_incidents(client: Client,
+                    last_run: Dict,
+                    first_fetch_time: str,
+                    max_fetch: str) -> Tuple[Dict[str, int], List[dict]]:
+    """
+    Fetch incidents (investigations) each minute (by default).
+
+    Args:
+        client(Client): Rapid7 Client
+        last_run(Dict[str, int]): Dict with last_fetch object,
+                                  saving the last fetch time(in millisecond timestamp)
+        first_fetch_time: Dict with first fetch time in str (ex: 3 days ago) need to be parsed
+        max_fetch(str): Max number of alerts per fetch. Defaults to 20, max is 1000
+    Returns:
+        Tuple of next_run (millisecond timestamp) and the incidents list
+    """
+    last_fetch_timestamp = last_run.get('last_fetch', None)
+
+    if last_fetch_timestamp:
+        last_fetch = datetime.fromtimestamp(last_fetch_timestamp / 1000)
+    else:
+        last_fetch, _ = parse_date_range(first_fetch_time)
+
+    incidents = []
+    next_run = last_fetch
+
+    size = int(max_fetch) if max_fetch else max_fetch
+    params = {'start_time': last_fetch.strftime(DATE_FORMAT),
+              'size': size}
+    investigations = client.list_investigations(remove_empty_elements(params))
+    for investigation in investigations.get('data', []):
+        investigation_created_time = investigation.get('created_time')
+
+        incident = {
+            'name': investigation.get('title'),
+            'occurred': investigation_created_time,
+            'rawJSON': json.dumps(investigation)
+        }
+        incidents.append(incident)
+
+        created_time = datetime.strptime(investigation_created_time, DATE_FORMAT)
+        if created_time > next_run:
+            next_run = created_time
+
+    # add 1 millisecond to next_run to prevent duplication
+    next_run = next_run + timedelta(milliseconds=1)
+    next_run_timestamp = int(datetime.timestamp(next_run) * 1000)
+
+    return {'last_fetch': next_run_timestamp}, incidents
+
+
+def main():
+    """PARSE AND VALIDATE INTEGRATION PARAMS"""
+
+    params = demisto.params()
+    region = params.get('region', {})
+    api_key = params.get('apiKey', {})
+    max_fetch = params.get('max_fetch', None)
```

Comment:
let's have a default value if not given.
```suggestion
    max_fetch = params.get('max_fetch', '50')
```